boilerplate/app/app.tsx

```diff
@@ -44,39 +55,34 @@ function App(props: AppProps) {
 
   const [areFontsLoaded] = useFonts(customFontsToLoad)
 
-  // Kick off initial async loading actions, like loading fonts and RootStore
-  useEffect(() => {
-    setupRootStore().then((rootStore) => {
-      setRootStore(rootStore)
+  const { rehydrated } = useInitialRootStore(() => {
+    // This runs after the root store has been initialized and rehydrated.
 
-      // If your initialization scripts run very fast, it's good to show the splash screen for just a bit longer to prevent flicker.
-      // Slightly delaying splash screen hiding for better UX; can be customized or removed as needed,
-      // Note: (vanilla Android) The splash-screen will not appear if you launch your app via the terminal or Android Studio. Kill the app and launch it normally by tapping on the launcher icon. https://stackoverflow.com/a/69831106
-      // Note: (vanilla iOS) You might notice the splash-screen logo change size. This happens in debug/development mode. Try building the app for release.
-      setTimeout(hideSplashScreen, 500)
-    })
-  }, [])
+    // If your initialization scripts run very fast, it's good to show the splash screen for just a bit longer to prevent flicker.
+    // Slightly delaying splash screen hiding for better UX; can be customized or removed as needed,
+    // Note: (vanilla Android) The splash-screen will not appear if you launch your app via the terminal or Android Studio. Kill the app and launch it normally by tapping on the launcher icon. https://stackoverflow.com/a/69831106
+    // Note: (vanilla iOS) You might notice the splash-screen logo change size. This happens in debug/development mode. Try building the app for release.
+    setTimeout(hideSplashScreen, 500)
+  })
```

Comment:
I like this pattern (as I mentioned above, the name could probably be improved). Open to feedback, however.

Reply:
I think this is good overall. Before, we had a bit simpler way to control when the components below would show by delaying setting the rootStore before certain items in the callback would finish (async ones in some cases). 

For example, you could have functions that kickoff async permissions checks or auth checks, then set the rootStore, then hide the splash screen.

As an alternative, I would see myself adding another useEffect to react to `rehydrated` to run certain things before setting a new `useState` variable that controls the visibility of below elements.

Reply:
Good point — in a way, there’s almost a whole “flow” that has to happen before we display the app.

Maybe we should expose an async function that really formalizes that whole flow. I’ll think about this a bit more today.